nmagent/requests.go

```diff
@@ -94,7 +94,7 @@ func (p *PutNetworkContainerRequest) Path() string {
 func (p *PutNetworkContainerRequest) Validate() error {
 	err := internal.ValidationError{}
 
-	if p.Version == 0 {
+	if p.Version == "0" {
```

Comment:
Need @timraymond 's help to understand why the version can't be `0`:
`2022/12/31 00:11:36 [1] [Azure-CNS] Failed to publish Network Container in put Network Conatainer call: 10cbe947-b86b-469b-a266-38207119e212 with err: building request: validating request: missing fields: Version`

It usually has been in the past..
`[CNS] Publish NC 02a39a95-d7fa-425e-b26b-fbde55be252e, nodeID: aks-podpool-18213534-vmss000000, body:CreateOrUpdateNetworkContainerRequest:[IPV4Addresses:192.168.0.4, SubnetName:delegated, Version:0, VirtualNetworkID:aee206a9-c130-426d-a76b-622b1972eb2b, VlanID:1, GreKey:511] `